From a6aac408c56112f73d28ea8567c29b2a7fe8fccc Mon Sep 17 00:00:00 2001
Message-Id: <a6aac408c56112f73d28ea8567c29b2a7fe8fccc.1649662980.git.geliang.tang@suse.com>
From: Jean-Philippe Brucker <jean-philippe@linaro.org>
Date: Mon, 8 Mar 2021 19:25:22 +0100
Subject: [PATCH] libbpf: Fix arm64 build
Git-commit: a6aac408c56112f73d28ea8567c29b2a7fe8fccc
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: No


The macro for libbpf_smp_store_release() doesn't build on arm64, fix it.

Fixes: 291471dd1559 ("libbpf, xsk: Add libbpf_smp_store_release libbpf_smp_load_acquire")
Signed-off-by: Jean-Philippe Brucker <jean-philippe@linaro.org>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20210308182521.155536-1-jean-philippe@linaro.org
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/lib/bpf/libbpf_util.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/lib/bpf/libbpf_util.h b/tools/lib/bpf/libbpf_util.h
index 94a0d7bb6f3c..cfbcfc063c81 100644
--- a/tools/lib/bpf/libbpf_util.h
+++ b/tools/lib/bpf/libbpf_util.h
@@ -35,7 +35,7 @@ extern "C" {
 		typeof(*p) ___p1;					\
 		asm volatile ("ldar %w0, %1"				\
 			      : "=r" (___p1) : "Q" (*p) : "memory");	\
-		__p1;							\
+		___p1;							\
 	})
 #elif defined(__riscv)
 # define libbpf_smp_store_release(p, v)					\
-- 
2.34.1

